speechbrain/core.py

```diff
@@ -1078,36 +1096,41 @@ def fit(
                             self._save_intra_epoch_ckpt()
                         last_ckpt_time = time.time()
 
-            # Run train "on_stage_end" on all processes
```

Comment:
What are all these changes on the core.py, I don't understand? This is an absolutely critical place on SB.